### PR TITLE
feat: detect installed version of framework

### PR DESCRIPTION
**Pull request by ericapisani**
#### Summary

Fixes [this issue](https://github.com/netlify/pod-ecosystem-frameworks/issues/240)

Currently one of the most common follow up questions we ask our customers is 'what version of the framework do you use?' when trying to debug an issue.

This change exposes that information as part of the`listFrameworks` method which is used by the [`build-info` project](https://github.com/netlify/build/tree/main/packages/build-info) as part of the [build process](https://github.com/netlify/buildbot/blob/3a000de5d3727998f3297394596d287d7e49aecd/bot/stage_detect_framework.go#L22-L37).

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/framework-info/issues/new/choose) before writing your code 🧑‍💻. This
      ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a
      typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: test/dev.js [hunk 1]
```diff
@@ -26,17 +26,6 @@ test('Should default dev.commands to framework.dev.command', async (t) => {
...
-test('Should return the same result when script order is different', async (t) => {
-  const frameworksBuildFirst = await getFrameworks('scripts-order/build-first')
-  const frameworksDevFirst = await getFrameworks('scripts-order/dev-first')
-
-  t.is(frameworksBuildFirst.length, 1)
-  t.is(frameworksDevFirst.length, 1)
-
-  t.deepEqual(frameworksBuildFirst[0].dev.commands, ['npm run dev', 'npm run start', 'npm run build'])
-  t.deepEqual(frameworksBuildFirst, frameworksDevFirst)
-})
```
Comment (ericapisani) [inline test/dev.js:38]: This test didn't make the most sense to me so I removed it

File: src/context.js [hunk 2]
```diff
@@ -1,29 +1,29 @@
-import { cwd, version } from 'process'
+import { cwd, version as nodejsVersion } from 'process'
```
Comment (maxcell) [inline src/context.js:1]: Love the change!

File: src/main.js [hunk 3]
```diff
@@ -19,16 +24,62 @@ import { listFrameworks as list, hasFramework as has, getFramework as get } from
...
  * @property {Build} build - Information about the build command
  * @property {object} env - Environment variables that should be set when calling the dev command
  * @property {string[]} plugins - A list of recommend Netlify build plugins to install for the framework
  */
 
+/**
+ * Gets the version of the framework that is installed in a project.
+ *
+ * This cannot currently be used in the browser at this time, which is why it's defined
+ * here rather than in `core.js` as part of the `getFrameworkInfo` method
```
Comment (maxcell) [inline src/main.js:48]: Love this note here for the rationale!

File: src/core.js [hunk 4]
```diff
@@ -157,6 +158,10 @@ const getFrameworkInfo = function (
   return {
     id,
     name,
+    package: {
+      name: detect.npmDependencies[0],
+      version: 'unknown',
```
Comment (nickytonline) [inline src/core.js:163]: What the reason `version` is set to `'unknown'`?

File: src/main.js [hunk 5]
```diff
@@ -19,16 +24,62 @@ import { listFrameworks as list, hasFramework as has, getFramework as get } from
  * @property {string} directory - Relative path to the directory where files are built
  */
 
+/**
```
Comment (nickytonline) [inline src/main.js:27]: This syntax works, but just FYI, that you can write them in a *.d.ts file as well. See https://github.com/netlify/plugins/blob/main/types/plugins.d.ts and where it gets referenced here https://github.com/netlify/plugins/blob/main/bin/sync_plugins_to_cms.js#L13

File: src/core.js (hunk 4 above)
Comment (ericapisani) [reply src/core.js:163]: I thought that `undefined` was not a great thing to output to the build logs in the event that there was an issue detecting the version, so I decided on this instead.

**Pull request merged**